substrate/primitives/weights/src/weight_meter.rs

```diff
@@ -51,13 +51,13 @@ pub struct WeightMeter {
 
 impl WeightMeter {
 	/// Creates [`Self`] from a limit for the maximal consumable weight.
-	pub fn from_limit(limit: Weight) -> Self {
+	pub fn with_limit(limit: Weight) -> Self {
 		Self { consumed: Weight::zero(), limit }
 	}
 
 	/// Creates [`Self`] with the maximal possible limit for the consumable weight.
-	pub fn max_limit() -> Self {
-		Self::from_limit(Weight::MAX)
+	pub fn new() -> Self {
```

Comment:
But the old one was more expressive?

Reply:
This comes after a discussion with @gavofyork around the readability of this piece of code:
```rust
let meter = WeightMeter::max_limit();
meter.consume();
```
It reads as though the meter starts from the limit and should saturate with the further `consume` call. Going through the impl makes it clear that it still starts from 0.

Since the limit to max is implied with a meter, we should not need to specify it explicitly.

Reply:
Maybe `with_max_limit`? I mean your example is understandable, but `new` is still not really expressive to what the limit will be. (I will also not block on this, just some thought)